components-api/src/main/scala/pl/touk/nussknacker/engine/api/context/ProcessCompilationError.scala

```diff
@@ -29,7 +29,7 @@ object ProcessCompilationError {
   val ValidatedNelApplicative: Applicative[ValidatedNelCompilationError] =
     Applicative[ValidatedNelCompilationError]
 
-  trait InASingleNode {
+  sealed trait InASingleNode {
```

Comment:
Is there a reason why the traits are not sealed and case classes not final here?

Reply:
Lately, we decided to make final, all case classes so you can add a final modifier always when you see that is missing. I thought that nobody was so crazy about extending case class so we could rely on the convention, but our new team members said that they saw cases of case class extension :) Regarding sealed, I'm a fan of the rule that you add this modifier when you want to do a pattern matching on all subclasses. I don't like the approach "let's add it because for now, it compiles". I think it is an unnecessary limitation. In the beginning, we had `StateStatus` sealed and it caused pain in the ass when we wanted to have some new status in our extension. We can talk more about props and cons if you don't feel convinced.

Reply:
I see now that we already use these classes in pattern matching so it is fine to add sealed :+1: